xarray/namedarray/core.py

```diff
@@ -812,5 +974,12 @@ def _to_dense(self) -> Self:
         else:
             raise TypeError("self.data is not a sparse array")
 
+    def _nonzero(self) -> tuple[Self, ...]:
+        """Equivalent to numpy's nonzero but returns a tuple of NamedArrays."""
+        # TODO we should replace dask's native nonzero
+        # after https://github.com/dask/dask/issues/1076 is implemented.
+        nonzeros = np.nonzero(self.data)
+        return tuple(type(self)((dim,), nz) for nz, dim in zip(nonzeros, self.dims))
+
```

Comment:
```suggestion
```

i believe we already have this in place, 

https://github.com/pydata/xarray/blob/eb749448f9513517230090b2cb0e3521ccb4f20f/xarray/namedarray/core.py#L756